central/tlsconfig/tlsconfig.go

```diff
@@ -183,3 +215,32 @@ func validForAllDNSNames(cert *x509.Certificate, dnsNames ...string) bool {
 	}
 	return true
 }
+
+var allowedAdditionalCAExtensions = map[string]bool{
+	".crt": true,
+	".pem": true,
+}
+
+var skipAdditionalCAFileMsg string
+
+func init() {
+	var allowedAdditionalCAExtensionsList = make([]string, 0, len(allowedAdditionalCAExtensions))
+	for ext := range allowedAdditionalCAExtensions {
+		allowedAdditionalCAExtensionsList = append(allowedAdditionalCAExtensionsList, ext)
+	}
+	sort.Strings(allowedAdditionalCAExtensionsList)
+	skipAdditionalCAFileMsg = "skipping additional-ca file %q because it has an invalid extension; allowed file extensions for additional ca certificates are "
+	for i, ext := range allowedAdditionalCAExtensionsList {
+		if i > 0 && i < len(allowedAdditionalCAExtensionsList)-1 {
+			skipAdditionalCAFileMsg += ", "
+		} else if i == len(allowedAdditionalCAExtensionsList)-1 {
+			skipAdditionalCAFileMsg += " and "
```

Comment:
You could use a string slice, append the skipped files and later explode the string, writing it to a log message.
No one likes to debug string concatenations (especially not if it results in grammar errors). :(

Reply:
I'm not sure I understand your recommendation fully. Do you mind giving an example ?

Reply:
> See comment for `strings.Join` above.

@SimonBaeumer  Yeah, I simplified the building of the error message a lot, basically defining the alllowed extensions in a slice and using this directly for the error message.